src/toxcore/crypto_core.rs

```diff
@@ -58,8 +58,8 @@ pub fn crypto_init() -> bool {
     //       `CRYPTO_INIT` *can't* be modified while it may be read by
     //       something else.
     CRYPTO_INIT_ONCE.call_once(|| {
-        let initialized = ::sodiumoxide::init();
-        unsafe { CRYPTO_INIT = initialized; }
+        ::sodiumoxide::init().expect("could not initialize libsodium");
```

Comment:
With this change `crypto_init` will always return `true`? Then better make it unit. btw, it seems we don't need this magic with `call_once` - `::sodiumoxide::init()` can be called multiple times and from different threads:
> sodium_init() initializes the library and should be called before any other function provided by Sodium. It is safe to call this function more than once and from different threads -- subsequent calls won't have any effects.

Reply:
Well, https://github.com/sodiumoxide/sodiumoxide/issues/223 . Let's fix it the next sodiumoxide release